### PR TITLE
Fixed #32873 -- Removed USE_L10N setting.

### PR DESCRIPTION
**Pull request by claudep**
ticket-32873.

### CODE REVIEW

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -807,7 +807,7 @@ def test_L10N_deactivated(self):
         Check if L10N is deactivated, the JavaScript i18n view doesn't
```
Comment (rfleschenberg) [inline tests/admin_views/tests.py:807]: This docstring needs updating, I think.

**Comment by carltongibson** [comment]
Hey @claudep — all positive on the mailing list. Want to progress this with a ticket and a _Ready for review_? Thanks!

**Comment by carltongibson** [comment]
@claudep — I created ticket-32873. Thanks!

**Comment by interDist** [comment]
would it be an idea to document the non-changeable nature (*) of the `floatformat` [template filter](https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#floatformat) as part of this PR?

(*) that the filter ignores any localisation setting and will always return a localised result after `USE_L10N` is removed. see https://code.djangoproject.com/ticket/30086.

**Comment by felixxm** [comment]
@claudep Is this ready for reviews? (it's marked as draft) I'd would like to include it in Django 4.0.

**Comment by claudep** [comment]
I'm a bit annoyed by the (small?) backwards incompatibility that suddenly `floatformat` will start producing localized numbers for users that had `USE_L10N` to `False`, and that we currently have no way to offer a way to disable that (as filters don't take context), except writing your own filter/tag, of course. What do you think?

**Comment by felixxm** [comment]
> I'm a bit annoyed by the (small?) backwards incompatibility that suddenly `floatformat` will start producing localized numbers for users that had `USE_L10N` to `False`, and that we currently have no way to offer a way to disable that (as filters don't take context), except writing your own filter/tag, of course. What do you think?

Maybe we can add a new "flag", e.g. `u` (similar to `g`), and document that `floatformat`'s output is now always localized, unless you pass `u` to turn off localization :thinking: This should also fix ticket-30086.

**Comment by claudep** [comment]
Wow, `floatformat` is going to get a suuuuper-charged filter :-) If we choose that route, I think we should also transition floatformat to a tag in parallel. But yes, that would be a solution.

**Comment by felixxm** [comment]
> Wow, `floatformat` is going to get a suuuuper-charged filter :-) If we choose that route, I think we should also transition floatformat to a tag in parallel. But yes, that would be a solution.

I prepared a draft, see #14832.

**Comment by interDist** [comment]
will this make it into the same release as #14832?

**Comment by felixxm** [comment]
> will this make it into the same release as #14832?

Yes, I'd like to include both in Django 4.0.

**Comment by felixxm** [comment]
@claudep Thanks :+1: I rebased and pushed small edits. I will add more `versionchanged` annotations later.

I wonder if there is a way to deprecate the `USE_L10N` setting and remove it in Django 5.0, is this feasible? Do we have arguments for removing it immediately? This seems a little against our stability policy :thinking: We can always request a vote of the Technical Board :ballot_box:  What do you think?

**Comment by claudep** [comment]
What would be the deprecation tip? Telling people using `USE_L10N=False` to set it to True?

**Comment by felixxm** [comment]
> What would be the deprecation tip? Telling people using `USE_L10N=False` to set it to True?

Yes, we could change the default in the global settings to `True` and raise a deprecation warning when `USE_L10N` is overridden. Maybe it's fine to remove it is immediately :thinking:, I'm not sure.

**Comment by claudep** [comment]
It is indeed difficult to estimate the impact of removing it now as this patch does on the projects in the wild. For sure, going the deprecation way is the safest move.

**Comment by felixxm** [comment]
@claudep I prepared PR #14519 based on this one but with a deprecation.

**Comment by claudep** [comment]
#14519 is this one :-)

**Comment by felixxm** [comment]
> #14519 is this one :-)

#14846 :smile:

**Comment by felixxm** [comment]
Closing in favor of #14846.

**Pull request closed without merging**